src/components/CustomFieldForm.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState } from 'react';
+import { initialValuesFormik, FormValues, yupSchemaCustomForm } from '../utils/yupSchema';
+import { useFormik } from 'formik';
+import { Box, Button } from '@chakra-ui/react';
+import { sendObj } from '../utils/fetchSupabase';
+import { SingleInputField } from './SingleInputField';
+import { createObject } from '../utils/functions';
+import { SingleSelectInput } from './SingleSelectInput';
+import { dishType } from '../utils/data';
+
+// (fields depending on the dish type should be required conditionally based on what type of dish is selected).
+// ma to tutaj sens jak ustawilem wartosci startowe w initialState?? to tez ulatwia walidacje i jupscheme
+
+// w teorii jak takie warunkowe wyświetlanie powinno wyglądać, gdyby initialState były puste i required we wszystkich oknach?
+
+export default function CustomFieldForm() {
+  const [iSsend, setIsSend] = useState(false);
+  const [isValidState, setIsValidState] = useState({ value: 'isLoading' });
+  const formik = useFormik<FormValues>({
+    enableReinitialize: true,
+    initialValues: initialValuesFormik,
+
+    onSubmit: () => {
+      setIsSend(true);
+      const newObj = createObject(formik.values);
+      const dataToServer = JSON.stringify(newObj);
+
+      sendObj(dataToServer).then((res) => setIsValidState(res));
```

Comment:
formik.values raczej zawsze są obiektem, (zobaczę później tę funkcję) ten JSON.stringify to musiałby być w tym sendObj, bo zawsze jak tam podasz dane to je trzeba zamienić na stringify, więc tam jest miejsce na tę logikę